azext_iot/product/command_tests.py

```diff
@@ -142,14 +146,125 @@ def initialize_workspace(cmd, product_name, working_folder="PnPCert"):
+        attestation_type,
+        certificate_path,
+        endorsement_key,
+        badge_type,
+        models)
+    return get_sdk(cmd).create_device_test(provisioning, body=test_configuration)
+
+
+def _build_test_configuration(product_id, device_type, attestation_type, certificate_path, endorsement_key, badge_type, models):
+    config = {
+        'validationType': 'Certification',
+        'productId': product_id,
+        'deviceType': device_type,
+        'provisioningConfiguration': {
+            'type': attestation_type
+        },
+        'certificationBadgeConfigurations': [
+            {
+                'type': badge_type
+            }
+        ]
+    }
+    if (attestation_type == AttestationType.symmetricKey.value):
+        config['provisioningConfiguration']['symmetricKeyEnrollmentInformation'] = {}
+    elif (attestation_type == AttestationType.tpm.value):
+        config['provisioningConfiguration']['tpmEnrollmentInformation'] = {
+            'endorsementKey': endorsement_key
+        }
+    elif (attestation_type == AttestationType.x509.value):
+        config['provisioningConfiguration']['x509EnrollmentInformation'] = {
+            'base64EncodedX509Certificate': _read_certificate_from_file(certificate_path)
+        }
+    if (badge_type == BadgeType.Pnp.value and models):
+        models_array = _process_models_directory(models)
+        config['certificationBadgeConfigurations'][0]['digitalTwinModelDefinitions'] = models_array
+
+    return config
+
+
+def _read_certificate_from_file(certificate_path):
+    with open(file=certificate_path, mode='rb') as f:
+        data = f.read()
+
+        from base64 import encodestring
+        return encodestring(data)
+
+
+def _process_models_directory(from_directory):
+    from azext_iot.common.utility import scantree, process_json_arg
+
+    models = []
+    for entry in scantree(from_directory):
+        if not any(
+            [
+                entry.name.endswith('.json'),
+                entry.name.endswith('.dtdl')
+            ]
+        ):
+            logger.debug(
+                'Skipping {} - model file must end with .json or .dtdl'.format(entry.path)
+            )
+            continue
+        entry_json = process_json_arg(content=entry.path, argument_name=entry.name)
+        # we need to double-encode the JSON string
+        from json import dumps
+        models.append(dumps(entry_json))
+    return models
+
+
+def _create_from_file(configuration_file):
+    if not (os.path.exists(configuration_file)):
+        raise CLIError("Specified configuration file does not exist")
+
+    # read the json file and POST /deviceTests
+    with open(file=configuration_file, encoding='utf-8') as f:
+        file_contents = f.read()
+
+        from json import loads
+
+        return loads(file_contents)
 
 
 def show(cmd, test_id):
     # call to GET /deviceTests/{deviceTestId}
-    return True
+    return get_sdk(cmd).get_device_test(device_test_id=test_id)
```

Comment:
We can simplify this one step further in the provider itself (not relevant to this particular PR, but just as an FYI) - here's an example of how we load the SDK in the PnP provider (ignore the suffix since we're currently targeting one single environment):
https://github.com/Azure/azure-iot-cli-extension/blob/f14c7f9759bc792359adae97e6ee33c712b211b6/azext_iot/pnp/providers/__init__.py#L45-L56

```
 class PnPModelRepositoryApiManager(object):
    def __init__(self, cmd):
        assert cmd
        self.cmd = cmd

    def get_mgmt_sdk(self, pnp_dns_suffix=None):
        return pnp_digitaltwin_modelrepo_api_service_factory(self.cmd, pnp_dns_suffix)
```

And then we can use it like this:
```
class ModelApiProvider(PnPModelRepositoryApiManager):
    def __init__(self, cmd, pnp_dns_suffix=None):
        super(ModelApiProvider, self).__init__(cmd=cmd)
        self.mgmt_sdk = self.get_mgmt_sdk(pnp_dns_suffix)
```
So our provider calls look like the following:
```
self.mgmt_sdk.get_model_async(
```

Reply:
I might do something similar in the next PR. That's pretty nice